packages/@aws-cdk/aws-cloudfront/lib/function.ts

```diff
@@ -0,0 +1,109 @@
+import { IResource, Names, Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnFunction } from './cloudfront.generated';
+
+/**
+ * Represents the function's source code
+ */
+export abstract class FunctionCode {
```

Comment:
Would add a `public static fromFile(path: string)` here

Reply:
Would definitely be a nice-to-have; not a blocker for approval, but certainly wouldn't hurt.

Reply:
I will add this later. I want to do this using an extra class for it and want to think about the API. I would love to have esbuild/bundling here but given that everything is ES5 this is not trivial.